ClearKeep/Scenes/Search/ViewModel/SearchGroupViewModel.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  SearchGroupViewModel.swift
+//  ClearKeep
+//
+//  Created by MinhDev on 26/07/2022.
+//
+
+import Foundation
+import Model
+import Networking
+import SwiftUI
+import simd
+
+protocol ISearchGroupViewModels {
+	var viewModelGroup: [SearchGroupViewModel] { get }
+}
+
+struct SearchGroupViewModels: ISearchGroupViewModels {
+	var viewModelGroup: [SearchGroupViewModel]
+	init(model: ISearchModels, responseUser: ISearchModels) {
+
+		let dataSearch = model.groupModel?.compactMap { data in
+			SearchGroupViewModel(model: data)
+		} ?? []
+		self.viewModelGroup = model.groupModel?.filter { $0.groupType == "group" }.compactMap { data in
+			SearchGroupViewModel(model: data)
+		} ?? []
+		let group = dataSearch.filter { $0.groupType == "group" }
+		let peer = dataSearch.filter { $0.groupType != "group" }
+
+		peer.forEach { users in
+			let user = users.groupMembers.filter { $0.userId != DependencyResolver.shared.channelStorage.currentServer?.profile?.userId }.compactMap { members in
+				SearchGroupViewModel(modelPeer: members, idGroup: users.groupId)
+			}
+			user.forEach { data in
+				viewModelGroup.append(data)
+			}
+		}
+
+		group.forEach { data in
+			let member = data.groupMembers.filter { $0.domain != DependencyResolver.shared.channelStorage.currentDomain
+			}.compactMap { members in
+				SearchGroupViewModel(modelMember: members)
+			}
+			member.forEach { data in
+				if !viewModelGroup.contains(where: { $0.userId == data.userId }) {
+					viewModelGroup.append(data)
+				}
+
+			}
+		}
+		
+		let userdata = responseUser.members?.compactMap { data in
+			SearchGroupViewModel(user: data, domain: DependencyResolver.shared.channelStorage.currentDomain)
+		}
+		userdata?.forEach { data in
+			if !viewModelGroup.contains(where: { $0.userId == data.userId }) {
+				viewModelGroup.append(data)
+			}
+		}
+	}
+}
+
+struct SearchGroupViewModel: IGroupModel {
+	var groupId: Int64 = 0
+	var groupName: String = ""
+	var groupAvatar: String = ""
+	var groupType: String = ""
+	var createdBy: String = ""
+	var createdAt: Int64 = 0
+	var updatedBy: String = ""
+	var updatedAt: Int64 = 0
+	var rtcToken: String = ""
+	var groupMembers: [IMemberModel] = []
+	var isJoined: Bool = false
+	var ownerDomain: String = ""
+	var ownerClientId: String = ""
+	var lastMessage: IMessageModel? = nil
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/8649f3f2f0ce9de1db3f8c96a6cf34266ff6e2d9/ClearKeep/Scenes/Search/ViewModel/SearchGroupViewModel.swift#L78"></span>Initializing an optional variable with nil is redundant.<br />
<code>redundant_optional_initialization</code> <code>SearchGroupViewModel.swift:78</code></td>
    </tr>
  </tbody>
</table>